Src/Support/Google.Apis.Auth/OAuth2/ServiceCredential.cs

```diff
@@ -157,9 +173,13 @@ public ServiceCredential(Initializer initializer)
         /// <inheritdoc/>
         public async Task InterceptAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
-            var accessToken = await GetAccessTokenForRequestAsync(request.RequestUri.AbsoluteUri, cancellationToken)
-                .ConfigureAwait(false);
-            AccessMethod.Intercept(request, accessToken);
+            var accessToken = await GetAccessTokenWithHeadersForRequestAsync(request.RequestUri.AbsoluteUri, cancellationToken).ConfigureAwait(false);
+            AccessMethod.Intercept(request, accessToken.AccessToken);
+
+            foreach (var header in accessToken.Headers)
```

Comment:
Should we have a method in AccessTokenWithHeaders that does this, rather than the duplicate code? Happy for that to be a separate PR.

Reply:
Ah yes, makes sense. Will add in a following PR.